src/Kestrel.Core/Internal/Http2/Http2OutputProducer.cs

```diff
@@ -74,7 +74,9 @@ public void Dispose()
             }
         }
 
-        public void Abort(ConnectionAbortedException abortReason)
+        // Review: This is called when a CancellationToken fires mid-write. In HTTP/1.x, this aborts the entire connection.
+        // Should we do that here?
+        void IHttpOutputAborter.Abort(ConnectionAbortedException abortReason)
```

Comment:
Reminder: On user induced cancellation call ResetAndAbort to make sure a reset is sent to the client and no graceful EndOfStream is sent. Otherwise it looks like a graceful truncation to the client which is a ship blocker.

Reply:
Right, otherwise the client has to depend on their own timeouts.

Reply:
#3007